assets/src/components/v2/subway_status/eink/subway_status.tsx

```diff
@@ -0,0 +1,312 @@
+import React, { ComponentType, forwardRef } from "react";
+import { classWithModifier, firstWord, imagePath } from "Util/util";
+import useTextResizer from "Hooks/v2/use_text_resizer";
+import {
+  Alert,
+  ContractedSection,
+  GLBranch,
+  GLMultiPill,
+  LineColor,
+  Section,
+  SubwayStatusData,
+  SubwayStatusPill,
+  adjustAlertForContractedStatus,
+  getAlertID,
+  isAlertLocationMap,
+  isContracted,
+  isContractedWith1Alert,
+  isExtended,
+  isGLMultiPill,
+} from "../subway_status_common";
+
+////////////////
+// COMPONENTS //
+////////////////
+
+const SubwayStatus: ComponentType<SubwayStatusData> = (props) => {
+  const { blue, orange, red, green } = props;
+
+  return (
+    <div className="subway-status">
+      {shouldShowHeader(props) && (
+        <span className="subway-status__header">Current Subway Service</span>
+      )}
+      <LineStatus section={blue} color={LineColor.Blue} />
+      <LineStatus section={orange} color={LineColor.Orange} />
+      <LineStatus section={red} color={LineColor.Red} />
+      <LineStatus section={green} color={LineColor.Green} />
+    </div>
+  );
+};
+
+type LineStatusProps = { section: Section; color: LineColor };
+
+const LineStatus: ComponentType<LineStatusProps> = ({ section, color }) => {
+  let alerts: [Alert] | [Alert, Alert] = isContracted(section)
+    ? section.alerts
+    : [section.alert];
+
+  const routePill = getRoutePillObject(section, color);
+  const showInlineBranches =
+    color === LineColor.Green &&
+    ((isContracted(section) && section.alerts.length > 1) ||
+      isExtended(section));
+
+  return (
+    <div className="subway-status_row">
+      <div className="subway-status_route-pill-container">
+        <SubwayStatusRoutePill routePill={routePill} />
+      </div>
+      <ContractedStatus
+        alerts={alerts}
+        showInlineBranches={showInlineBranches}
+      />
+      <div className="subway-status_status_rule" />
+    </div>
+  );
+};
+
+const getRoutePillObject = (
```

Comment:
I'm using this Miro as my frame of reference, and it seems like the routePill should never be more complicated than just the line name. Why is there so much logic here and in `SubwayStatusRoutePill` to calculate the big route pill?

Reply:
If you look at [this Figma file](https://www.figma.com/file/TjbJshbl7fk63GDebLokmu/Widgets-%E2%80%94-Subway-status?type=design&node-id=345-26902&t=wXqe84q2C22dPaKT-4), there is an update to GL pill that isn't reflected in the Miro board. If there is only one GL alert and it is not a trunk alert/affects the whole line, we show a GL + branch icon combo similar to LCD. If the alert affects the trunk/whole line or there are more than one alerts, we show the GL pill and inline the branch icons next to the alert.